backend/dal/Helpers/Extensions/LeaseExtensions.cs

```diff
@@ -25,7 +25,8 @@ private static IQueryable<Entity.Lease> GenerateCommonLeaseQuery(this IQueryable
             filter.ThrowIfNull(nameof(user));
 
             if (!String.IsNullOrWhiteSpace(filter.TenantName)) {
-                query = query.Where(l => l.Tenant != null && EF.Functions.Like(l.Tenant.Surname + ", " + l.Tenant.FirstName + ", " + l.Tenant.MiddleNames, $"%{filter.TenantName}%"));
+                query = query.Where(l => l.Persons.Any(person => person != null && EF.Functions.Like(person.Surname + ", " + person.FirstName + ", " + person.MiddleNames, $"%{filter.TenantName}%"))
```

Comment:
This ensures that updates to the schema do not break the pre-existing L&L search functionality.